rmw_opensplice_cpp/src/rmw_publisher.cpp

```diff
@@ -170,6 +170,13 @@ rmw_create_publisher(
 
   publisher->implementation_identifier = opensplice_cpp_identifier;
   publisher->data = publisher_info;
+  publisher->topic_name = reinterpret_cast<const char *>(
+    rmw_allocate(sizeof(char) * strlen(topic_name) + 1));
+  if (!publisher->topic_name) {
+    RMW_SET_ERROR_MSG("failed to allocate memory for node name");
+    goto fail;
+  }
+  memcpy(const_cast<char *>(publisher->topic_name), topic_name, strlen(topic_name) + 1);
```

Comment:
Regarding the above discussion about multiplying by sizeof(char), it seems like if we switch to using strcpy down here it makes sense to leave out to sizeof(char) here as well (but to keep it in the above rmw_allocate call).

I don't know much about platforms where sizeof(char) != 1 (seems very rare/nonstandard?) but it seems like `str` functions would be implemented with that in mind and the unit for the len argument for `strn` functions would account for that...